### DISCUSSION TITLE
How to create Events for test fixtures?

### DISCUSSION THREAD

**Discussion by daniel-hampton** [arlyon/async-stripe#441] [Q&A] [answered]
I'm trying to write tests for a function that processes Stripe events. Running into a problem creating test fixture data for these event types: https://docs.rs/async-stripe/0.23.0/stripe/struct.Event.html

At the moment, I'm stuck on how to even create a mock EventID. https://docs.rs/async-stripe/0.23.0/stripe/struct.EventId.html

I see in the repo there's a test that uses a mock client, but this seems complicated for what I'm trying to do. https://github.com/arlyon/async-stripe/blob/master/tests/checkout.rs

Any advice?

**Comment by arlyon** [comment, marked as answer]
You may be interested in this (stalled) PR https://github.com/arlyon/async-stripe/pull/374

If you'd like to open a new one with the same contents I will try to get it merged for you.

**Reply by arlyon** [reply]
Note that for this you still need a real signature, which you could get for example from triggering one using the stripe CLI

**Reply by daniel-hampton** [reply]
thanks for the reply. I'll look at the PR.

Someone pointed me at these part of the code where it's creating IDs by deserializing with serde. Which makes sense.
https://docs.rs/async-stripe/0.23.0/src/stripe/ids.rs.html#642

Might work for my use case.
